pkg/plugins/backendplugin/grpcplugin/client.go

```diff
@@ -55,7 +71,7 @@ func newClientConfig(executablePath string, args []string, env []string, skipHos
 		Logger:           logWrapper{Logger: logger},
 		AllowedProtocols: []goplugin.Protocol{goplugin.ProtocolGRPC},
 		GRPCDialOptions: []grpc.DialOption{
-			grpc.WithStatsHandler(otelgrpc.NewClientHandler()),
+			grpc.WithStatsHandler(otelgrpc.NewClientHandler(otelgrpc.WithTracerProvider(newClientTracerProvider(tracer)))),
```

Comment:
It would be good to add a comment here as to why this is necessary IE why this dial option has to be this way as opposed to the other places where we can get away with just `otelgrpc.NewClientHandler()` 🙏